features/location/impl/src/main/kotlin/io/element/android/features/location/impl/SendLocationView.kt

```diff
@@ -0,0 +1,142 @@
+import io.element.android.libraries.ui.strings.CommonStrings
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
+@Composable
+fun SendLocationView(
+    state: SendLocationState,
+    modifier: Modifier = Modifier,
+    onBackPressed: () -> Unit = {},
+) {
+    val mapState = rememberMapState()
+    BottomSheetScaffold(
+        sheetContent = {
+            Spacer(modifier = Modifier.height(16.dp))
+            ListItem(
+                headlineContent = {
+                    Text(stringResource(CommonStrings.screen_share_this_location_action))
+                },
+                modifier = Modifier.clickable {
+                    state.eventSink(
+                        SendLocationEvents.ShareLocation(
+                            lat = mapState.position.lat,
+                            lng = mapState.position.lon
+                        )
+                    )
+                    onBackPressed()
+                },
+                leadingContent = {
+                    Icon(Icons.Default.LocationOn, null)
+                },
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+        },
+        modifier = modifier,
+        scaffoldState = rememberBottomSheetScaffoldState(
+            bottomSheetState = rememberStandardBottomSheetState(initialValue = SheetValue.Expanded),
+        ),
+        sheetDragHandle = {},
+        sheetSwipeEnabled = false,
+        topBar = {
+            CenterAlignedTopAppBar(
+                title = {
+                    Text(
+                        text = stringResource(CommonStrings.screen_share_location_title),
+                        fontSize = 16.sp,
+                        fontWeight = FontWeight.SemiBold,
+                    )
+                },
+                navigationIcon = {
+                    BackButton(onClick = onBackPressed)
+                },
+            )
+        },
+    ) {
+        Box(
+            modifier = Modifier
+                .padding(it)
+                .consumeWindowInsets(it),
+            contentAlignment = Alignment.Center
+        ) {
+            MapView(
+                modifier = Modifier.fillMaxSize(),
+                mapState = mapState,
+            )
+            Icon(
+                resourceId = R.drawable.pin,
+                contentDescription = null,
+                tint = Color.Unspecified
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+internal fun SendLocationViewLightPreview(@PreviewParameter(SendLocationStateProvider::class) state: SendLocationState) =
+    ElementPreviewLight { ContentToPreview(state) }
+
+@Preview
+@Composable
+internal fun SendLocationViewDarkPreview(@PreviewParameter(SendLocationStateProvider::class) state: SendLocationState) =
+    ElementPreviewDark { ContentToPreview(state) }
+
+@Composable
+private fun ContentToPreview(state: SendLocationState) {
+    SendLocationView(
+        state = state,
+        onBackPressed = {},
+    )
```

Comment:
Hmm, where are the screenshot tests for these? Do you need to run the action? :)

(I guess they're mostly placeholders anyway, but :shrug:)

Reply:
Yeah, action started now!